### PR TITLE
chore: Tooling Updates

### PR DESCRIPTION
**Pull request by cpheps**
### Proposed Change
Updated all tool versions and GitHub Action versions.

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,20 +9,20 @@ jobs:
...
           cache-dependency-path: '**/go.sum'
       - name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v4
+        uses: goreleaser/goreleaser-action@v5
         with:
           # either 'goreleaser' (default) or 'goreleaser-pro'
           distribution: goreleaser-pro
-          version: "v1.18.2"
+          version: "v1.22.1"
           args: build --single-target --skip-validate --clean
```
Comment (BinaryFissionGames) [inline .github/workflows/release.yml:26]:
```suggestion
          version: "v1.22.1"
          args: build --single-target --skip=validate --clean
```

**Review by BinaryFissionGames — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -70,20 +70,20 @@ jobs:
...
```
Comment (BinaryFissionGames) [inline .github/workflows/release.yml:87]:
```suggestion
          version: "v1.22.1"
          args: build --single-target --skip=validate --clean
```

File: .github/workflows/manual_msi_build.yml [hunk 3]
```diff
@@ -71,18 +71,18 @@ jobs:
...
           check-latest: true
       - name: Run GoReleaser
-        uses: goreleaser/goreleaser-action@v4
+        uses: goreleaser/goreleaser-action@v5
         with:
           # either 'goreleaser' (default) or 'goreleaser-pro'
           distribution: goreleaser-pro
-          version: "v1.18.2"
+          version: "v1.22.1"
           args: build --single-target --skip-validate --clean --snapshot
```
Comment (BinaryFissionGames) [inline .github/workflows/manual_msi_build.yml:86]: I might have missed more of these 😄 
```suggestion
          version: "v1.22.1"
          args: build --single-target --skip=validate --clean --snapshot
```

Comment (cpheps) [reply .github/workflows/manual_msi_build.yml:86]: Good catch, I'll double check them all.

**Review by jsirianni — APPROVED** [review]

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**